core/src/main/scala/kafka/cluster/Partition.scala

```diff
@@ -647,15 +647,18 @@ class Partition(val topic: String,
 
   /**
     * @param leaderEpoch Requested leader epoch
-    * @return The last offset of messages published under this leader epoch.
+    * @return The requested leader epoch and the last offset of messages published under this
```

Comment:
It looks like the existing Java doc (prior to this patch) of this method is not correct. According to Java doc of LeaderEpochFileCache.endOffsetFor(...), it says `The End Offset is the end offset of this epoch, which is defined as the start offset of the first Leader Epoch larger than the Leader Epoch requested, or else the Log End Offset if the latest epoch was requested`.

Reply:
Yes, I think the prior description is more of a shortcut, which is actually not correct. I just realized that we should use "end offset" instead of the last offset of messages published here -- the description in LeaderEpochFileCache is more precise. I will update this comment accordingly.